### PR TITLE
Fix propogating structure argument in s3Cluster

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix propogating structure argument in s3Cluster. Previously the `DEFAULT` expression of the column could be lost when sending the query to the replicas in s3Cluster.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit d7aaf053f992d4d6b27e9f88133818bb0a48d011 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/69147/d7aaf053f992d4d6b27e9f88133818bb0a48d011/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/1c839ff4b95570cb324cfd9307a0825c32fc50c7/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Review by antonio2368 — APPROVED** [review]

**Comment by Avogar** [comment]
I reworked the code a bit. Also fixed one more [issue](https://github.com/ClickHouse/clickhouse-core-incidents/issues/198#issue-2493371161)

File: src/Storages/ObjectStorage/Azure/Configuration.h [hunk 1]
```diff
@@ -22,13 +22,39 @@ class StorageAzureConfiguration : public StorageObjectStorage::Configuration
 
     static constexpr auto type_name = "azure";
     static constexpr auto engine_name = "Azure";
+    /// All possible signatures for Azure engine with structure argument (for example for azureBlobStorage table function).
+    static constexpr auto max_number_of_arguments_with_structure = 4;
```
Comment (antonio2368) [inline src/Storages/ObjectStorage/Azure/Configuration.h:26]: Should it be 7 and 8?

**Comment by clickgapai** [comment]
Hi @Avogar @antonio2368 — a couple of suggestions to make this PR easier for reviewers and the changelog reader.

**Description** (currently `missing`):
- Why: The PR body is the raw template with no explanation of the bug, root cause, or what changed.
- Suggested addition:

  > Explain: cluster table functions (s3Cluster/hdfsCluster/azureBlobStorageCluster) previously appended only the inferred `structure` when propagating arguments to remote nodes, and for named collections did not propagate an inferred `format`. This caused remote nodes to re-infer format/structure and could produce mismatched schemas. The fix appends both `format` and `structure` (only overriding values that are 'auto') and unifies per-engine argument-count validation via `getSignatures`/`getMaxNumberOfArguments`.

Ignore this if the current title/body is intentional.

**Pull request merged**